metadata_updater/metadata_updater.py

```diff
@@ -0,0 +1,333 @@
+import logging
+
+try:
+    import log
+except:
+    from . import log
+
+logger = logging.getLogger(__name__)
+ERRORS = 0
+
+class ConfigReader():
+    """
+    Create a config object that can
+    be passed to the scripts methods
+    """
+
+    def __init__(self):
+        # READ CONFIG
+        with open(os.getcwd()  + '/config.yaml', 'r') as f:
+            config = yaml.load(f)
+
+        if "Connection" in config:
+            self.api_key = config["Connection"]["Api_key"]
+            self.domain = config["Connection"]["Domain"]
+        else:
+
+            raise SystemExit('CONFIG ERROR: No "Connection" section')
+
+        if "Text" in config:
+            self.text_update = config["Text"]["Update"]
+        else:
+            raise SystemExit('CONFIG ERROR: No "Text" section')
+
+        if "Output" in config:
+            self.destination_dir = config["Output"]["Destination"]
+        else:
+            self.destination_dir = os.getcwd()  + '/'
+
+        if "Datasets" in config:
+            self.layers = config["Datasets"]["Layers"]
+            self.filter = config["Datasets"]["Filter"]
+        else:
+            raise SystemExit('CONFIG ERROR: No "Datasets" section')
+
+        if "Test" in config:
+            self.test_dry_run = config["Test"]["Dry_run"]
+            if self.test_dry_run not in (True, False):
+                raise SystemExit('CONFIG ERROR: "Test Dry Run" must be ' \
+                '"True" or "False". Got:"{}" instead'.format(self.test_dry_run))
+
+            self.test_overwrite = config["Test"]["Overwrite_files"]
+            if self.test_overwrite not in (True, False):
+                raise SystemExit('CONFIG ERROR: "test_overwrite" must be ' \
+                '"True" or "False". Got:"{}" instead'.format(self.test_overwrite))
+        else:
+            raise SystemExit('CONFIG ERROR: No "Test" section')
+
+def post_metadata(draft, file):
+    """
+    Update the Data Service draft version for the 
+    layer with edited metadata
+    """
+
+    global ERRORS
+
+    try:
+        xml = open(file).read()
+        draft.set_metadata(xml.encode('utf-8'))
+        return True
+    except koordinates.exceptions.ServerError as e:
+        if 'Expected 204 response, got 200' in str(e):
+            #This is a known bug. - Bug#3
+            return True
+        else:
+            ERRORS += 1
+            logger.critical('metadata update for {0} fail with {1}'.format(draft.id,
+                                                                            str(e)))
+            return False
+
+def get_metadata(layer, config):
+    """
+    Download the layers metadata file
+    """
+
+    file_destination = os.path.join(config.destination_dir,'{0}_{1}.iso.xml'.format(layer.type, 
+                                                                                    layer.id))
+    if config.test_overwrite:
+        file_exists(file_destination)
+            
+    layer.metadata.get_xml(file_destination)
```

Comment:
Tried doing a dry run to retrieve all CC metadata files. It successfully retrieved 1938 files but then had this error at the end.

'''
Traceback (most recent call last):
  File "metadata_updater.py", line 333, in <module>
    main()
  File "metadata_updater.py", line 304, in main
    file = get_metadata(layer, config)
  File "metadata_updater.py", line 109, in get_metadata
    layer.metadata.get_xml(file_destination)
AttributeError: 'NoneType' object has no attribute 'get_xml'

'''

Reply:
I managed to reproduce this. see #15. This was due to 504 responses. I have built in retires and if it still fails I log out the details and request the user to review this at the scripts completion